File: apps/kbve/kbve-react-native/modules/kbve-wgpu/android/src/main/java/expo/modules/kbvewgpu/KbveWgpuModule.kt
package expo.modules.kbvewgpu

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class KbveWgpuModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("KbveWgpuModule")

    Function("setJwt") { jwt: String ->
      KbveWgpuView.current?.setJwt(jwt)
    }

    Function("goOnline") { serverUrl: String, jwt: String ->
      KbveWgpuView.current?.goOnline(serverUrl, jwt)
    }

    Function("hostResponse") { id: Int, ok: Boolean, payload: String ->
      KbveWgpuView.current?.hostResponse(id, ok, payload)
    }

    Function("pointer") { kind: Int, x: Double, y: Double ->
      KbveWgpuView.current?.pointer(kind, x, y)
    }

    View(KbveWgpuView::class) {
      Events("onReady", "onHostCall")
      Prop("componentId") { view: KbveWgpuView, value: String ->
        view.componentId = value
      }
    }
  }
}

File: apps/kbve/kbve-react-native/modules/kbve-wgpu/ios/KbveWgpuModule.swift
import ExpoModulesCore

public class KbveWgpuModule: Module {
    public func definition() -> ModuleDefinition {
        Name("KbveWgpuModule")

        Function("setJwt") { (jwt: String) in
            KbveWgpuView.current?.setJwt(jwt)
        }

        Function("goOnline") { (serverUrl: String, jwt: String) in
            KbveWgpuView.current?.goOnline(serverUrl, jwt)
        }

        Function("hostResponse") { (id: Int, ok: Bool, payload: String) in
            KbveWgpuView.current?.hostResponse(id: id, ok: ok, payload: payload)
        }

        Function("pointer") { (kind: Int, x: Double, y: Double) in
            KbveWgpuView.current?.pointer(kind, x, y)
        }

        View(KbveWgpuView.self) {
            Events("onReady", "onHostCall")
            Prop("componentId") { (view: KbveWgpuView, value: String) in
                view.componentId = value
            }
        }
    }
}
